haystack/document_stores/elasticsearch.py

```diff
@@ -2012,27 +2002,59 @@ def _get_vector_similarity_query(self, query_emb: np.ndarray, top_k: int):
 
     def _scale_embedding_score(self, score):
         # adjust approximate knn scores, see https://opensearch.org/docs/latest/search-plugins/knn/approximate-knn
-        if self.embeddings_field_supports_similarity:
-            if self.similarity == "dot_product":
-                if score > 1:
-                    score = score - 1
-                else:
-                    score = -(1 / score - 1)
-            elif self.similarity == "cosine":
-                score = -(1 / score - 2)
-            elif self.similarity == "l2":
-                score = 1 / score - 1
         # adjust exact knn scores, see https://opensearch.org/docs/latest/search-plugins/knn/knn-score-script/
-        else:
-            if self.similarity == "dot_product":
-                raise Exception("Exact dot_product similarity is not supported.")
-            elif self.similarity == "cosine":
+        if self.similarity == "dot_product":
+            if score > 1:
+                score = score - 1
+            else:
+                score = -(1 / score - 1)
+        elif self.similarity == "l2":
+            score = 1 / score - 1
+        elif self.similarity == "cosine":
+            if self.embeddings_field_supports_similarity:
+                score = -(1 / score - 2)
+            else:
                 score = score - 1
-            elif self.similarity == "l2":
-                score = 1 / score - 1
 
         return score
 
+    def _clone_embedding_field(
```

Comment:
Where is this method called?

Reply:
Not at all. It's just a helper function that can be useful if you want to switch the similarity function with ANN support and without having to recalculate all embeddings. If it's confusing more than helping I'm happy to drop it.